src/snowflake/snowpark/modin/plugin/_internal/index.py

```diff
@@ -844,10 +844,10 @@ def copy(
         >>> idx is new_idx
         False
         """
-        # TODO: SNOW-1458120 implement copy
-        WarningMessage.index_to_pandas_warning("copy")
         return Index(
-            self.to_pandas().copy(deep=deep, name=name), convert_to_lazy=self.is_lazy
+            self._query_compiler.copy(),
```

Comment:
This is going to be an confusing APIs, it seems when the index is local index, the copy is configurable with data copy or not, but when the index is lazy, the copy can never be configured with deep or not.
@sfc-gh-azhan @sfc-gh-samdesai1 I have a clarification question, when the index is lazy, and when the index is local? is all index created with data like pd.Index([1, 2, 3]) local, and all index created from a snowpark pandas object remote?